packages/table/src/Table.tsx

```diff
@@ -0,0 +1,196 @@
+`;
+
+interface TableProps extends React.ComponentPropsWithoutRef<'table'> {
+  data?: Array<any>;
+  columns?:
+    | Array<ReactElement<HeaderRowProps>>
+    | Array<ReactElement<TableHeaderProps> | string>;
+  selectable?: boolean;
+}
+
+export default function Table({
+  columns = [],
+  data = [],
+  selectable: selectableProp = false,
+  children,
+  ...rest
+}: TableProps) {
+  const initialState: State = {
+    sort: {
+      direction: 'asc',
+    },
+    data,
+    selectable: selectableProp,
+    headerCheckState: false,
+    headerIndeterminate: false,
+  };
+
+  const [state, dispatch] = useReducer(reducer, initialState);
+
+  React.useEffect(() => {
+    const rowCheckedState: {
+      [n in number]: undefined | boolean;
+    } = {};
+
+    rows.forEach((datum, index) => {
+      if (datum.props.disabled) {
+        return;
+      }
+      rowCheckedState[index] = undefined;
+    });
+
+    if (rowCheckedState) {
+      dispatch({
+        type: Types.SetRowCheckedState,
+        payload: rowCheckedState,
+      });
+    }
+  }, [data]);
+
+  React.useEffect(() => {
+    if (state.rowCheckedState) {
+      const boolArray = Object.values(state.rowCheckedState);
+      const checkSame = boolArray.every(val => val === boolArray[0]);
+
+      if (checkSame) {
+        dispatch({
+          type: Types.ToggleHeaderIndeterminate,
+          payload: false,
+        });
+
+        if (boolArray[0]) {
+          dispatch({
+            type: Types.ToggleHeaderCheckedState,
+            payload: true,
+          });
+        } else {
+          dispatch({
+            type: Types.ToggleHeaderCheckedState,
+            payload: false,
+          });
+        }
+      } else {
+        dispatch({
+          type: Types.ToggleHeaderIndeterminate,
+          payload: true,
+        });
+      }
+    }
+  }, [state.data, state.rowCheckedState]);
+
+  let usingHeaderRow = React.useMemo(() => false, [children]);
+  let rows: Array<React.ReactElement>;
+
+  if (typeof children === 'function') {
+    rows = data.map((datum, index) => children({ datum, index }));
+  }
+
+  const sortRows = (columnId: number, key: string) => {
+    dispatch({
+      type: Types.SortTableData,
+      payload: {
+        columnId,
+        key,
+        data,
+      },
+    });
+  };
+
+  const renderHeader = (array: Array<any>): React.ReactNode => {
+    const cols = array.map((child, index) => {
+      if (isComponentType(child, 'HeaderRow')) {
+        usingHeaderRow = true;
+
+        const { children } = child?.props;
+
+        const renderedChildren = Array.isArray(children)
+          ? children
+          : [children];
+
+        return React.cloneElement(child, {
+          children: renderHeader(renderedChildren),
+        });
+      }
+
+      if (isComponentType(child, 'TableHeader')) {
+        const { label, accessor: accessorProp } = child.props;
+
+        let glyph = 'Unsorted';
+
+        const accessor = accessorProp || label?.toLowerCase();
```

Comment:
I thought we'd do `accessorProp || key`? You're always going to at least need a key so it's doubtful we'd ever fall back to the label lowercased.

Also `accessProp` needs to support being a function so the table supports sorting nested objects. `accessor={data => data.whatever.foo}`.

Reply:
Updated this, let me know what you think.

Reply:
When using `key` here React throws: 

```Warning: TableHeader: `key` is not a prop. Trying to access it will result in `undefined` being returned. If you need to access the same value within the child component, you should pass it as a different prop. (https://fb.me/react-special-props)```